### DISCUSSION TITLE
Is the width and height of recorded videos configurable?

### DISCUSSION THREAD

**Discussion by unikitty37** [cypress-io/cypress#14250] [Questions and Help] [answered]
Cypress records its videos at 1280x720. This makes it a bit difficult to see some of the longer error messages in the left pane.

Is there a way to configure the width and height of the recorded videos?

I can only find config options for the viewport itself, which I want to leave at 1050x1000, but recorded in a video of 1600x1050 rather than 1280x720.

**Comment by jennifer-shehane** [comment, marked as answer]
The video is recorded in a resolution of the actual size of the screen (or virtual screen size when headless). The default size for headless is 1280x720. 

To set the screen size to a different size during headless runs, it is different for each headless browser. You can follow these instructions: https://on.cypress.io/browser-launch-api#Set-screen-size-when-running-headless

**Reply by unikitty37** [reply]
Thanks — I thought I'd looked at all the relevant parts of the docs, but I'd missed that bit 😁

**Reply by unikitty37** [reply]
OK, this is weird — I'm trying to view a load of debug output I'm dumping to the screen, so I have this in plugins/index.js temporarily:

```js
on('before:browser:launch', (browser = {}, launchOptions) => {
  if (browser.name === 'electron' && browser.isHeadless) {
    launchOptions.preferences.width = 1800
    launchOptions.preferences.height = 3400
  }

  return launchOptions
})
```

but it changes the size from 1280x720 to 1278x1022 instead of 1800x3400. Is there a hard limit that I'm running up against?

**Reply by jennifer-shehane** [reply]
There may be some limit with ffmpeg recording size. What size are screenshots when this is set?

**Reply by unikitty37** [reply]
They were coming out as 1279x1023, so it doesn't seem to be ffmpeg to blame (apart from chopping off the odd pixel — ISTR the dimensions have to be even…)

**Reply by jennifer-shehane** [reply]
Yeah, I've seen some people mention a 1px off issue with ffmpeg. I believe there are some workarounds for passing the ffmpeg size in this issue: https://github.com/cypress-io/cypress/issues/587 But overall, I'd like this to be more easily configurable.

**Reply by unikitty37** [reply]
Thanks — it seems that `launchOptions.preferences.resizable = false` is the key. Once I added that, the window was the expected size. I have no idea why that affects it, though!
